### ISSUE TITLE
New rule: `no-missing-route-params-for-links`

### ISSUE THREAD

**Issue by runspired** [ember-template-lint/ember-template-lint#2418]
Ideally the rule would be configurable to account for helpers like `href-to` `url-for` etc.

The problem I've encountered is that the following is valid Ember code:

In route.js

```ts
this.route("product", { path: `/product/:productId/:organizationId` }, function() {
  this.route("index");
});
```

```hbs
<LinkTo "product.index">Home</LinkTo>
<a href={{url-for "product.index"}}>Home</LinkTo>
```

In both of these scenarios ember is able to build the full url by looking up the model for the `product` route. However, this requires the product route to have been entered, and (currently) breaks in the helper form when the model changes since no arg is subscribed to (this can be fixed by accessing `router.currentRouteUrl` in the helper, but while that workaround should be added, this behavior seems too magical anyway).

I would link for the correct style to be

```hbs
<LinkTo "product.index" @models={{array this.currentProductId this.currentOrganizationId}}>Home</LinkTo>
<a href={{url-for "product.index" @models={{array this.currentProductId this.currentOrganizationId}}>Home</LinkTo>
```
